src/odemis/util/synthetic.py

```
@@ -681,7 +683,9 @@ def simulate(self, current_pos: dict) -> numpy.ndarray:
                 with numpy.errstate(all="raise"):
                     self._last_img = self._get_ray_traced_pattern(dx, dy, dz)
                 self._last_pos = current_pos.copy()
+                self._last_res = res
             except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)